app/routes.py

```diff
@@ -1,2 +1,153 @@
+        goal_data = request.get_json()
+        goal.title = goal_data["title"]
+        db.session.commit()
+        return make_response(goal.goal_data_structure(), 200)
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit() 
+        goal_response = {"details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'}
+        return make_response(goal_response, 200)
+    
+@goals_bp.route("/<goal_id>/tasks", methods=["GET", "POST"])
+def handle_goal_tasks(goal_id):
+    goal = Goal.query.get_or_404(goal_id)
+    if request.method == "GET":
+        # Method that appends full task info
+        return make_response(goal.tasks_data_structure(), 200)
+    if request.method == "POST":
+        request_body = request.get_json()
+        for task_id in request_body["task_ids"]:
+            task = Task.query.get(task_id)
+            task.goal_id = goal_id
+            db.session.commit()
+        return make_response({"id": goal.goal_id, "task_ids": request_body["task_ids"]}, 200)
+
+
+####################################################################
+#                              TASK ROUTES                        #
+####################################################################
+
+@tasks_bp.route("", methods=["GET", "POST"])
+def handle_tasks():
+    if request.method == "GET":
+        sort_query = request.args.get("sort")
+        if sort_query == "asc": 
+            tasks = Task.query.order_by(Task.title)
+        elif sort_query == "desc":
+            tasks = Task.query.order_by(Task.title.desc())
+        else:
+            tasks = Task.query.all()
+            
+        tasks_response = []
+        for task in tasks:
+            tasks_response.append(task.task_data_structure()['task'])
+
+        return make_response(jsonify(tasks_response), 200)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if 'title' in request_body and 'description' in request_body and 'completed_at' in request_body: 
+            new_task = Task(title=request_body["title"],
+                        description=request_body["description"],
+                        completed_at=request_body['completed_at'])
+            db.session.add(new_task)
+            db.session.commit()
+            return make_response(new_task.task_data_structure(), 201)
+        else:
+            return make_response({"details":"Invalid data"}, 400)
+
+
+@tasks_bp.route("/<task_id>", methods=["GET", "PUT", "DELETE"])
+def handle_task(task_id):
+    task = Task.query.get_or_404(task_id)
+    if request.method == "GET":
+        return make_response(task.task_data_structure(), 200)
+    elif request.method == "PUT":
+        task_data = request.get_json()
+        task.title = task_data["title"]
+        task.description = task_data["description"]
+        task.completed_at = task_data['completed_at']
+        db.session.commit()
+        task = Task.query.get(task_id)
```

Comment:
small thing, but we could keep from sending another query request again the route by sending the already modified `task` variable from before line 124 into the return request instead.

it saves our route a little power